vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc

```diff
@@ -724,6 +729,39 @@ void AdsImpl::OnPageLoaded(
       << previous_tab_url_;
 }
 
+void AdsImpl::ExtractPurchaseIntentSignal(const std::string& url) {
+  if (!ShouldClassifyPagesIfTargeted()) {
+    return;
+  }
+
+  // TODO(Terry Mancey): might fail TLD+1
```

Comment:
Please remove comment as I can now confirm the below code to match the URLs is correctly using TLD+1